packages/common/src/converter/search-attribute-payload-converter.ts

```diff
@@ -0,0 +1,63 @@
+import { ValueError } from '@temporalio/internal-workflow-common';
+import { PayloadConverter } from './payload-converter';
+import { JsonPayloadConverter } from './json-payload-converter';
+import { Payload, str } from './types';
+
+const jsonConverter = new JsonPayloadConverter();
+const validNonDateTypes = ['string', 'number', 'boolean'];
+
+/**
+ * Converts Search Attribute values using JsonPayloadConverter
+ */
+export class SearchAttributePayloadConverter implements PayloadConverter {
+  public toPayload(values: unknown): Payload | undefined {
+    if (!(values instanceof Array)) {
+      throw new ValueError(`SearchAttribute value must be an array`);
+    }
+
+    const firstValue = values[0];
```

Comment:
If you really want to be accurate with the error messages you could check that the array isn't empty or better yet, does empty array unset the value? Is unsetting supported?